### PR TITLE
Feature/ idva5 1172 new tech debt change delete endpoint

### PR DESCRIPTION
**Pull request by hemerem-ch**
__Short description outlining key changes/additions__

__JIRA Ticket Number__

### Type of change

* [ ] Bug fix
* [ ] New feature
* [ ] Breaking change
* [ ] Infrastructure change

### Pull request checklist

* [ ] I have added unit tests for new code that I have added
* [ ] I have added/updated functional tests where appropriate
* [ ] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/java.md)

__An exhaustive list of peer review checks can be read [here](https://github.com/companieshouse/styleguides/blob/master/java_review.md#developer-actions-prior-to-code-commitreview-started)__

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by sampankumar — APPROVED** [review]

File: src/main/java/uk/gov/companieshouse/acsp/service/AcspService.java [hunk 1]
```diff
@@ -209,4 +209,15 @@ public ResponseEntity<Object> deleteAcspApplication(String id) {
...
         }
     }
+
+    public ResponseEntity<Object> deleteAcspApplicationAndTransaction(String id, String transactionId) {
+        try {
+            acspRepository.deleteById(id);
+            transactionService.deleteTransaction(transactionId);
+            return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+        } catch (Exception e) {
+            LOGGER.error("Error deleting document with id " + id + "and transaction ID: " + transactionId, e);
```
Comment (cbroadley-ch) [inline src/main/java/uk/gov/companieshouse/acsp/service/AcspService.java]: Might need to add whitespace between the id + "and transaction ID: " on this log message.

Something like this?
`"Error deleting document with id: " + id + " and transaction ID: " + transactionId, e`

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by cbroadley-ch — APPROVED** [review]

**Review by sampankumar — APPROVED** [review]

**Pull request merged**